lib/src/utils/types_utils.dart

```diff
@@ -32,6 +32,95 @@ extension Subtypes on DartType {
     final element = this.element;
     return element is InterfaceElement ? element.allSupertypes : [];
   }
+
+  /// Formats the type string based on nullability and presence of generics.
+  String getTypeString({
+    required bool withGenerics,
+    required bool withNullability,
+  }) {
+    final displayString = getDisplayString(withNullability: withNullability);
+
+    return withGenerics ? displayString : displayString.replaceGenericString();
+  }
+
+  /// Checks if a variable type is among the ignored types.
+  bool hasIgnoredType({required Set<String> ignoredTypes}) {
+    if (ignoredTypes.isEmpty) return false;
+
+    final checkedTypes = [this, ...supertypes];
+
+    ignoredTypes = getIntersectionTypesFor(ignoredTypes: ignoredTypes);
```

Comment:
Don't mutate argument variables.

```suggestion
    final intersectionIgnoredTypes = getIntersectionTypesFor(ignoredTypes: ignoredTypes);
```